### PR TITLE
Flush collection contents on each DB unlock in API.

### PR DESCRIPTION
**Pull request by smira**
See #343

### CODE REVIEW

**Comment by smira** [comment]
@vincentbernat could you please take a look?

File: api/api.go [hunk 1]
```diff
@@ -39,38 +62,12 @@ func cacheFlusher(requests chan int, acks chan error) {
...
-				defer func() {
-					requests <- RELEASEDB
-					<-acks
-				}()
+			// if aptly API runs in -no-lock mode,
+			// caches are flushed when DB is closed anyway, no need
+			// to flush them
+			if requests == nil {
+				flushColections()
 			}
```
Comment (vincentbernat) [inline api/api.go]: No need to keep enclosing anonymous function since we don't need a scope anymore.

**Comment by vincentbernat** [comment]
LGTM, but not tested.

File: api/api.go (hunk 1 above)
Comment (smira) [inline api/api.go]: yeah, thanks, I'll fix it

**Pull request merged**